### PR TITLE
Updating Puppiproducer to include fillDescriptions

### PR DESCRIPTION
**Pull request by alefisico**
#### PR description:

To test PUPPI at HLT for Run3/PhaseII, PuppiProducer has to be updated with the fillDescription function to be recognized in condfb. This PR introduces this function in that producer. This change should be completely transparent to the PUPPI algorithm itself, or any other part of CMSSW.

#### PR validation:

Run `runTheMatrix.py -l limited -i all --ibeos`, and all the test passed. Here is the output of it: https://cernbox.cern.ch/index.php/s/QbCjBhba83NjFCr

### CODE REVIEW

File: CommonTools/PileupAlgos/plugins/PuppiPhoton.cc [hunk 1]
```diff
@@ -21,7 +21,7 @@
 #include "DataFormats/Common/interface/Association.h"
 #include "DataFormats/Math/interface/deltaR.h"
 //Main File
-#include "CommonTools/PileupAlgos/plugins/PuppiPhoton.h"
+#include "CommonTools/PileupAlgos/interface/PuppiPhoton.h"
```
Comment (perrotta) [inline CommonTools/PileupAlgos/plugins/PuppiPhoton.cc]: Why moving the headers to ```/interface```?
Please, bring them back to ```\plugin```

**Comment by slava77** [comment]
is there a master version of this?

File: CommonTools/PileupAlgos/plugins/PuppiPhoton.cc (hunk 1 above)
Comment (alefisico) [reply CommonTools/PileupAlgos/plugins/PuppiPhoton.cc]: done

**Comment by perrotta** [comment]
> is there a master version of this?

Indeed, I didn't notice it, thank you Slava...
@alefisico please provide a master version of this PR, which has to be merged before being able to integrate this backport to 10_6

**Comment by alefisico** [comment]
sorry for the stupid question, but I think I dont understand what do you mean by "master version"

**Comment by perrotta** [comment]
> sorry for the stupid question, but I think I dont understand what do you mean by "master version"

10_6_X is a version closed to developments.
All developments must be first integrated in the "master version", currently CMSSW_11_0_X, and only when merged in the master they can be considered for a possible backport to the closed versions, if justified (in this case, it is).

Please prepare a pull request for the master (select "base master" when you submit the PR in github, not 11_0_X, i.e. do not change the default) based on one of the latest 11_0_X IBs. When that PR will get merged, then we can consider merging also this backport PR.

**Comment by alefisico** [comment]
ok I just include the master #28020

**Comment by perrotta** [comment]
-1
(This will have to be modified anyhow once the review of #28020 is finished, all fixes suggested are integrated, and that master version of this PR will be ready to get merged. Let "-1" it for the moment.)

**Comment by alefisico** [comment]
This is a backport version of this PR #28020

**Comment by perrotta** [comment]
-1
Please either rebase correctly, or prepare a brand new PR with the backport

**Comment by kpedro88** [comment]
-1
please follow http://cms-sw.github.io/tutorial-resolve-conflicts.html#backporting-a-pr to make a proper backport

**Pull request closed without merging**